packages/ra-core/src/core/NavigateToFirstResource.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Navigate } from 'react-router';
+import { useFirstResourceWithListAccess } from './useFirstResourceWithListAccess';
+import { useCreatePath } from '../routing';
+
+/**
+ * This component will inspect the registered resources and navigate to the first one for which users have access to the list page.
+ * @param props
+ * @param props.loading The component to display while the component is loading.
+ */
+export const NavigateToFirstResource = ({
+    loading: LoadingPage,
+}: NavigateToFirstResourceProps) => {
+    const { resource, isPending } = useFirstResourceWithListAccess();
+    const createPath = useCreatePath();
+
+    if (isPending) {
+        return <LoadingPage />;
+    }
+
+    if (resource) {
+        return (
+            <Navigate
+                to={createPath({
+                    resource,
+                    type: 'list',
+                })}
+            />
+        );
+    }
```

Comment:
I think we should throw an error if there is no accessible resource.

Reply:
Should we redirect to `/authentication-error` or `/access-denied`?

Reply:
We don't know if the problem comes from a lack of resource or a lack of accessible resource, so I would just throw an error and let the error boundary to its job.